src/components/MessagesList/MessagesGroup/Message/MessagePart/AudioPlayer.vue

```diff
@@ -72,6 +80,37 @@ export default {
 		},
 	},
 
+	mounted() {
+		document.addEventListener('audioPlayerEnded', (e) => {
+			if (!this.$refs.audioPlayer) {
+				return
+			}
+
+			const voiceMessages = this.$store.getters.getVoiceMessages()
+
+			const previouslyPlayedVoiceMessage = voiceMessages[e.detail.id]
+			const thisVoiceMessage = voiceMessages[previouslyPlayedVoiceMessage.nextId]
+
+			if (previouslyPlayedVoiceMessage.nextId === this.id && thisVoiceMessage && thisVoiceMessage.since === 0) {
```

Comment:
In Whatsapp (whose example is given in the issue), the voice messages auto-play only when they are grouped together. Therefore I tried to replicate that only. If there's a message between two voice messages, it will not be auto-played.

Therefore this `thisVoiceMessage.since === 0` condition is given. If we don't want this behaviour we can remove this and it will auto-lay irrespective of they are grouped or not.

Reply:
WhatsApp (and others, I believe) plays not only grouped voice messages, but consequent ones, so from different authors, if they go one by one

Reply:
@Antreesy Yeah I meant that..

Just to confirm suppose a conversation is like this -

```
- Voice message 1
- Voice message 2
- Some text
- Voice message 3
```

Suppose we play voice message 1.
- Voice message 2 will auto-play.
- Will voice message 3 auto-play or not?

If the answer is yes, then I think we need to use a state to keep the voice messages mapping as I have done (mentioning as you said that we don't need a new state here - https://github.com/nextcloud/spreed/pull/13360#discussion_r1769582994)

```
[
    <id>: {
         id: <id>,
         nextID: <nextID>,
         prevID: <prevID>,
    },
    ....,
    ....,
]
```

And if you say that no, it shouldn't be auto-played as we have a text message in middle, then we don't need this state and we can do some computation on the `messageList` state we have to just check if this audio message is next to the previously played audio message as you said.

Please let me know your opinion!

Reply:
Did a quick test for two messengers I am using:
- WhatsApp stops at non-voice message, if present;
- Telegram does not stop at all until the end of chat.

I don't know which way would be preferred, we'll discuss it on working day (probably the first one, to not break the flow of discussion, and don't play voice message posted after several days, before you read everything between them). But as I see, both methods don't require a state management:
- a-la WhatsApp - get id of next message in the list, and if voice - play it;
- a-la Telegram - find next voice message in the list and play it.

Reply:
@Antreesy Got it! I have kept it in Whatsapp's style for now. Once it is decided let me know if it needs to be changed to Telegram's style.

> But as I see, both methods don't require a state management

Ahh I got it, what you are saying.

But I have a question related to optimization here - the one I have written vs the one you are proposing -

The one I have written -

- The voice mapping is pre-computed, in an "object".
- Therefore when we want to know the next voice message of a particular voice message.
- We just get the next voice message in O(1) time-complexity.

The one you are proposing - If we use `messagesList` state and compute it on audio play end -

- The `messagesList` state is an "array" not an object.
- Therefore first we have to find the current ID in that list in O(n) time complexity.
- And after that -
  - If its by Whatsapp's style, then we don't need to iterate any further and just check the next message.
  - But if its Telegram's style, then we have to keep looping "till" we find next voice message.
- Isn't it becoming too expensive (especially, when we have a long conversation list)?

Please let me know your opinion! 👍

Reply:
Hi, we appreciate your contribution :). I'd go with Whatsapp way of playing audio.

It's true that your method provides O(1) when looking for next playable audio, yet the initial computation and the state management of different elements have a few chances of being used (multiple audio messages sent in a row is not a daily case). Thus, it's preferrable to use O(n) in lookup rather than storing extra/duplicated data in store that we probably won't use at least the whole of it.

Reply:
Makes sense @DorraJaouad ! Updated the logic to use `messagesList` state and removed the `voiceMessages` related states I introduced. 👍

Reply:
I'd simplify that further, given that we're checking only the next message after the current one:

```js
	nextVoiceMessageId() {
		const messages = this.$store.state.messagesStore.messages
		const currentIndex = Object.keys(messages).indexOf(this.messageId)
		
		if (currentIndex === -1) {
			return null
		}
		
		const nextMessage = messages[currentIndex + 1]
		if (nextMessage?.messageType === 'voice-message') {
			return nextMessage.id
		} else {
			return null
		}
	}
```

Reply:
@Antreesy I forgot to put the comment, but I made this to make it a little flexible, but do you think this is better?

Reply:
Yes, we can always extend it later, if required. But for now, I'd keep changes as little as possible

P.S. My bad to not remember it earlier, but we already have nextMessageId in Message component, so no need to calculate it again:

https://github.com/nextcloud/spreed/blob/main/src%2Fcomponents%2FMessagesList%2FMessagesGroup%2FMessage%2FMessage.vue#L171-L174

Reply:
@Antreesy Okkk! Then its very simple now. Made the changes! 👍